service/config_provider.go

```diff
@@ -76,6 +76,27 @@ type configProvider struct {
 	watcher chan error
 }
 
+// Deprecated: use NewConfigProvider instead
+// MustNewConfigProvider returns a new ConfigProvider that provides the configuration:
+// * Retrieve the config.Map by merging all retrieved maps from all the configmapprovider.Provider in order.
+// * Then applies all the ConfigMapConverterFunc in the given order.
+// * Then unmarshalls the final config.Config using the given configunmarshaler.ConfigUnmarshaler.
+//
+// The `configMapProviders` is a map of pairs <scheme,Provider>.
+//
+// Notice: This API is experimental.
+func MustNewConfigProvider(
```

Comment:
If you want to go this path, you need to "deprecate" the current `NewConfigProvider` and in 2 weeks we change that to return error.

Same for the "NewDefault", need a must version and "deprecate"

Reply:
I deprecated it [here](https://github.com/open-telemetry/opentelemetry-collector/pull/4734/commits/22eb74f37ccd026ae0d8090575658a7e0bb57844)

The current PR was supposed to be the second step of the three given in this [comment](https://github.com/open-telemetry/opentelemetry-collector/pull/4734/commits/22eb74f37ccd026ae0d8090575658a7e0bb57844). It will be undraft after releasing version with the previous PR merged.

Also, sorry for an oversight at `NewDefaultConfigProvider` I'm going to push the commit with the must version to the [previous PR](https://github.com/open-telemetry/opentelemetry-collector/pull/4734).